nscplugin/src/main/scala/scala/scalanative/nscplugin/NirCodeGen.scala

```diff
@@ -157,12 +157,23 @@ abstract class NirCodeGen
       def pinned =
         if (!isModule(sym) || isExternModule(sym)) Seq()
         else Seq(Attr.PinAlways(genMethodName(sym.asClass.primaryConstructor)))
-
-      Attrs.fromSeq(
-          pinned ++ sym.annotations.collect {
+      val attrs = sym.annotations.collect {
         case ann if ann.symbol == ExternClass => Attr.Extern
         case ann if ann.symbol == PureClass   => Attr.Pure
-      })
+        case ann if ann.symbol == LinkClass =>
+          val Apply(_, args) = ann.tree
+          val values = args.map {
+            case Literal(Constant(str: String)) => str
+          }
+
+          values match {
+            case Seq(n)            => Attr.LinkDynamic(n)
+            case Seq(n, "dynamic") => Attr.LinkDynamic(n)
+            case Seq(n, "static")  => Attr.LinkStatic(n)
+          }
```

Comment:
What if I write `@link("something", "foo")`? You probably don't want to crash. You should report a compile error and gracefully recover with the default.